src/encoded/upgrade/dataset.py

```diff
@@ -731,3 +731,49 @@ def experiment_35_36(value, system):
     # https://encodedcc.atlassian.net/browse/ENCD-5964
     if value.get('assay_term_name') == 'Capture Hi-C':
         value['assay_term_name'] = 'capture Hi-C'
+
+
+@upgrade_step('reference', '20', '21')
+def reference_20_21(value, system):
+    # https://encodedcc.atlassian.net/browse/ENCD-6134
+    new_elements_selection_method = []
+    if 'elements_selection_method' in value:
+        for element_selection_method in value['elements_selection_method']:
+            if re.match('ATAC-seq', element_selection_method):
+                new_elements_selection_method.append('accessible genome regions')
+                if 'notes' in value:
```

Comment:
I think you need to define notes outside of the for loop like your new elements selection method array, because there could be multiple new notes added. Then at the end you only have to check if there is already a notes existing in the value once.